File: MainGame/LuaLib/Assembly-CSharp/Goods_EquipPutReturnProto.lua
---@class Goods_EquipPutReturnProto : System.ValueType
---@field public IsSuccess boolean
---@field public MsgCode number
---@field public Type number
---@field public GoodsId number
---@field public GoodsServerId number
---@field public ProtoCode number
---@field public ProtoEnName string
local m = {}

---@overload fun(): @virtual
---@virtual
---@param isChild boolean
---@return string
function m:ToArray(isChild) end

---@overload fun(buffer:string): @static
---@static
---@param buffer string
---@param isChild boolean
---@return Goods_EquipPutReturnProto
function m.GetProto(buffer, isChild) end

Goods_EquipPutReturnProto = m
return m

File: Server/GameServer/GameServerApp/GameServerApp/Proto/Goods_EquipPutReturnProto.cs
//===================================================
//作    者：边涯  http://www.u3dol.com
//创建时间：2019-11-13 22:42:00
//备    注：
//===================================================
using System.Collections;
using System.Collections.Generic;
using System;

/// <summary>
/// 服务器返回穿戴消息
/// </summary>
public struct Goods_EquipPutReturnProto : IProto
{
    public ushort ProtoCode { get { return 16013; } }
    public string ProtoEnName { get { return "Goods_EquipPutReturn"; } }

    public bool IsSuccess; //是否成功
    public int MsgCode; //消息码
    public byte Type; //0=穿上 1=脱下
    public int GoodsId; //装备编号
    public int GoodsServerId; //装备服务器端编号

    public byte[] ToArray(MMO_MemoryStream ms, bool isChild = false)
    {
        ms.SetLength(0);
        if (!isChild)
        {
            ms.WriteUShort(ProtoCode);
        }

        ms.WriteBool(IsSuccess);
        if (IsSuccess)
        {
            ms.WriteByte(Type);
            ms.WriteInt(GoodsId);
            ms.WriteInt(GoodsServerId);
        }
        else
        {
        }
        ms.WriteInt(MsgCode);

        return ms.ToArray();
    }

    public static Goods_EquipPutReturnProto GetProto(MMO_MemoryStream ms, byte[] buffer)
    {
        Goods_EquipPutReturnProto proto = new Goods_EquipPutReturnProto();
        ms.SetLength(0);
        ms.Write(buffer, 0, buffer.Length);
        ms.Position = 0;

        proto.IsSuccess = ms.ReadBool();
        if (proto.IsSuccess)
        {
            proto.Type = (byte)ms.ReadByte();
            proto.GoodsId = ms.ReadInt();
            proto.GoodsServerId = ms.ReadInt();
        }
        else
        {
        }
        proto.MsgCode = ms.ReadInt();

        return proto;
    }
}